<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>TADCompare: Identification and characterization of differential TADs • TADCompare</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="TADCompare: Identification and characterization of differential TADs">
<meta property="og:description" content="
    TADCompare is an R package designed to identify and characterize differential Topologically Associated 
    Domains (TADs) between multiple Hi-C contact matrices. It contains 
    functions for finding differential TADs between two datasets, finding
    differential TADs over time and identifying consensus TADs across multiple
    matrices. It takes all of the main types of HiC input and returns simple,
    comprehensive, easy to analyze results.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">TADCompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/TADCompare.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Input_Data.html">Input data formats</a>
    </li>
    <li>
      <a href="articles/Ontology_Analysis.html">Gene Ontology Enrichment Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dozmorovlab/TADCompare/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<p>Cresswell, Kellen G., and Mikhail G. Dozmorov. “<a href="https://doi.org/10.3389/fgene.2020.00158">TADCompare: An R Package for Differential and Temporal Analysis of Topologically Associated Domains</a>.” Frontiers in Genetics 11 (March 10, 2020): 158.</p>
<p><code>TADCompare</code> is an R package for differential Topologically Associated Domain (TAD) detection between two Hi-C contact matrices and across a time course, and TAD boundary calling across multiple Hi-C replicates. It has three main functions, <code>TADCompare</code> for differential TAD analysis, <code>TimeCompare</code> for time course analysis, and <code>ConsensusTADs</code> for consensus boundary identification. The <code>DiffPlot</code> function allows for visualizing the differences between two contact matrices.</p>
<div id="installation" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1></div>
<pre><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages(c('dplyr', 'PRIMME', 'cluster', 'Matrix', 'magrittr', 'HiCcompare'))

if (!requireNamespace("BiocManager", quietly=TRUE))
    install.packages("BiocManager")</a></code></pre>
<p>The latest version of <code>TADCompare</code> can be directly installed from Github:</p>
<pre><code><a href="https://devtools.r-lib.org//reference/remote-reexports.html">devtools::install_github('cresswellkg/TADCompare', build_vignettes = TRUE)
library(TADCompare)</a></code></pre>
<p>Alternatively, the package can be installed from Bioconductor (to be submitted):</p>
<pre><code>if (!requireNamespace("BiocManager", quietly=TRUE))
    install.packages("BiocManager")
BiocManager::install("TADCompare", version = "devel")
library(TADCompare)</code></pre>
</div>
<div id="input" class="section level1">
<h1 class="hasAnchor">
<a href="#input" class="anchor"></a>Input</h1>
<p>There are three types of input accepted:</p>
<ol>
<li>n x n contact matrices</li>
<li>n x (n+3) contact matrices</li>
<li>3-column sparse contact matrices</li>
</ol>
<p>It is required that the same format be used for each of the inputs to a given function, or an error will occur. These formats are explained in depth in the <a href="https://htmlpreview.github.io/?https://github.com/dozmorovlab/TADCompare/blob/master/vignettes/Input_Data.html">Input Data vignette</a>.</p>
</div>
<div id="usage" class="section level1">
<h1 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h1>
<p>Please, refer to the <a href="https://htmlpreview.github.io/?https://github.com/dozmorovlab/TADCompare/blob/master/vignettes/TADCompare.html">TADCompare vignette</a> for an in depth tutorial.</p>
<div id="tadcompare" class="section level2">
<h2 class="hasAnchor">
<a href="#tadcompare" class="anchor"></a>TADcompare</h2>
<p>Differential TAD detection (<code>TADCompare</code>) involves identifying differing TAD boundaries between two contact matrices. Accordingly, the input is the two contact matrices that we would like to find these boundaries in and their corresponding resolution. The output is a dataset containing boundary scores for all regions and a dataset containing all differential and non-differential TAD boundaries.</p>
<pre><code><a href="https://rdrr.io/r/utils/data.html"># Load example contact matrices
data("GM12878.40kb.raw.chr2")
data("IMR90.40kb.raw.chr2")

# Find differential TADs
TD_Compare &lt;- TADCompare(GM12878.40kb.raw.chr2, IMR90.40kb.raw.chr2, resolution = 40000)</a></code></pre>
<p>We can then print the set of regions with at least one TAD boundary:</p>
<pre><code>head(TD_Compare$TAD_Frame)

  Boundary  Gap_Score TAD_Score1 TAD_Score2     Differential Enriched_In             Type
1  8200000 -2.0245884 0.02537611   1.596684     Differential    Matrix 2             &lt;NA&gt;
2  8240000  0.8091967 2.08454863   1.258245 Non-Differential    Matrix 1 Non-Differential
3  8880000  2.1191814 3.44450158   1.471221     Differential    Matrix 1            Split
4  8960000 -1.6590363 0.46678509   1.712167 Non-Differential    Matrix 2 Non-Differential
5  9560000 -2.3232567 0.56031218   2.313139     Differential    Matrix 2            Merge
6  9600000  0.2375074 1.91851750   1.552303 Non-Differential    Matrix 1 Non-Differential</code></pre>
<p>And, visualize a specific region on a chromosome:</p>
<pre><code><a href="reference/DiffPlot.html"># Visualizing the results
DiffPlot(tad_diff    = TD_Compare, 
         cont_mat1   = GM12878.40kb.raw.chr2,
         cont_mat2   = IMR90.40kb.raw.chr2,
         resolution  = 40000,
         start_coord = 8000000,
         end_coord   = 16000000,
         show_types  = TRUE, 
         point_size  = 5,
         palette     = "RdYlBu",
         rel_heights = c(1, 2))</a></code></pre>
<p><img src="https://mdozmorov.github.io/BIOS691.2018/assets/plot_original.png"></p>
<p><code>TADCompare</code> detects TAD boundaries by selecting regions with TAD boundary scores above a threshold (1.5 by default). An alternative way of running <code>TADCompare</code> is to call TAD boundaries using a separate TAD caller, and then compare those pre-defined TAD boundaries. The example below uses the <a href="https://bioconductor.org/packages/devel/bioc/html/SpectralTAD.html">SpectralTAD</a> TAD caller to pre-define TAD boundaries.</p>
<pre><code># Call TADs using SpectralTAD
bed_coords1 = bind_rows(SpectralTAD(GM12878.40kb.raw.chr2, chr = "chr2", levels = 3))
bed_coords2 = bind_rows(SpectralTAD(IMR90.40kb.raw.chr2,   chr = "chr2", levels = 3))
# Placing the data in a list for the plotting procedure
Combined_Bed = list(bed_coords1, bed_coords2)

# Running TADCompare with pre-specified TADs
TD_Compare &lt;-  TADCompare(GM12878.40kb.raw.chr2, IMR90.40kb.raw.chr2, resolution = 40000, pre_tads = Combined_Bed)

# Visualizing the results
DiffPlot(tad_diff    = TD_Compare, 
         cont_mat1   = GM12878.40kb.raw.chr2,
         cont_mat2   = IMR90.40kb.raw.chr2,
         resolution  = 40000,
         start_coord = 8000000,
         end_coord   = 16000000,
         pre_tad     = Combined_Bed,
         show_types  = FALSE, 
         point_size  = 5,
         palette     = "RdYlBu",
         rel_heights = c(1, 1))</code></pre>
<p><img src="https://mdozmorov.github.io/BIOS691.2018/assets/plot_predefined.png"></p>
</div>
<div id="timecompare" class="section level2">
<h2 class="hasAnchor">
<a href="#timecompare" class="anchor"></a>TimeCompare</h2>
<p><code>TimeCompare</code> takes data from at least four time points and identifies all regions with at least one TAD. Using this information, it then classifies each region, based on how they change over time, into 6 categories (Dynamic, Highly Common, Early Appearing/Disappearing, and Late Appearing/Disappearing).</p>
<pre><code><a href="https://rdrr.io/r/utils/data.html">data("time_mats")
Time_Mats = TimeCompare(time_mats, resolution = 50000)
head(Time_Mats$TAD_Bounds)</a></code></pre>
<p>The resulting output is:</p>
<pre><code>  Coordinate   Sample 1   Sample 2   Sample 3   Sample 4 Consensus_Score               Category
1   16900000 -0.6733709 -0.7751516 -0.7653696 15.1272253     -0.71937026     Late Appearing TAD
2   17350000  3.6406563  2.3436229  3.0253018  0.7840556      2.68446232 Early Disappearing TAD
3   18850000  0.6372268  6.3662245 -0.7876844  6.9255446      3.50172563    Early Appearing TAD
4   20700000  1.5667926  3.0968633  2.9130479  2.8300136      2.87153075            Dynamic TAD
5   22000000 -1.0079676 -0.7982571  0.6007264  3.1909178     -0.09876534     Late Appearing TAD
6   22050000 -1.0405532 -0.9892242 -0.2675822  4.2737511     -0.62840320     Late Appearing TAD</code></pre>
<p>For each coordinate, we have the individual boundary score for each sample (Sample x), consensus boundary score (Consensus_Score), and category (Category).</p>
</div>
<div id="consensustads" class="section level2">
<h2 class="hasAnchor">
<a href="#consensustads" class="anchor"></a>ConsensusTADs</h2>
<p><code>ConsensusTADs</code> uses a novel metric called the consensus boundary score to identify TAD boundaries consistently defined across multiple contact matrices. It can operate on an unlimited number of replicates, time points, or conditions.</p>
<pre><code><a href="https://rdrr.io/r/utils/data.html">data("time_mats")
con_tads = ConsensusTADs(time_mats, resolution = 50000)
head(con_tads$Consensus)</a></code></pre>
<pre><code> Coordinate  Sample 1 Sample 2   Sample 3 Sample 4 Consensus_Score
1   18850000 0.6372268 6.366224 -0.7876844 6.925545        3.501726
2   28450000 3.1883107 3.313883  3.1711743 3.913620        3.251097
3   30000000 2.0253285 3.477652  2.9314321 3.926354        3.204542
4   32350000 2.6978488 2.455860  3.5131909 3.550942        3.105520
5   36900000 3.0731406 3.153978  3.1861296 4.489285        3.170054</code></pre>
<p>The results are a set of coordinates with significant consensus TADs. Columns starting with “Sample” refer to the individual boundary scores. Consensus_Score is the consensus boundary score across all samples.</p>
</div>
<div id="downstream-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#downstream-analysis" class="anchor"></a>Downstream analysis</h2>
<p>The output of <code>TADcompare</code> and <code>TimeCompare</code> functions may be used for a range of analyses on genomic regions. One common one is gene ontology enrichment analysis to determine the pathways in which genes near TAD boundaries occur in. An example is shown in the <a href="https://htmlpreview.github.io/?https://github.com/dozmorovlab/TADCompare/blob/master/vignettes/Ontology_Analysis.html">Ontology_Analysis vignette</a></p>
</div>
</div>
<div id="availability" class="section level1">
<h1 class="hasAnchor">
<a href="#availability" class="anchor"></a>Availability</h1>
<p>The developmental version is available at <a href="https://github.com/cresswellkg/TADCompare" class="uri">https://github.com/cresswellkg/TADCompare</a>, the stable version is available at <a href="https://github.com/dozmorovlab/TADCompare" class="uri">https://github.com/dozmorovlab/TADCompare</a>. The <code>master</code> branch contains code that can be installed into the current R version 3.6 and above. The <code>Bioconductor</code> branch contains code with the <code>Depends: R (&gt;= 4.0)</code> requirement needed for the Bioconductor submission.</p>
</div>
<div id="contributions-and-support" class="section level1">
<h1 class="hasAnchor">
<a href="#contributions-and-support" class="anchor"></a>Contributions and Support</h1>
<p>Suggestions for new features and bug reports are welcome. Please, create a new issue for any of these or contact the author directly: @cresswellkg (cresswellkg[at]vcu[dot]edu)</p>
</div>
<div id="contributors" class="section level1">
<h1 class="hasAnchor">
<a href="#contributors" class="anchor"></a>Contributors</h1>
<p>Authors: @cresswellkg (cresswellkg[at]vcu[dot]edu) &amp; @mdozmorov (mikhail.dozmorov[at]vcuhealth[dot]org)</p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/TADCompare">https://​www.bioconductor.org/​packages/​TADCompare</a>
</li>
<li>Browse source code at <br><a href="https://github.com/dozmorovlab/TADCompare/">https://​github.com/​dozmorovlab/​TADCompare/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/dozmorovlab/TADCompare/issues">https://​github.com/​dozmorovlab/​TADCompare/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing TADCompare</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Kellen Cresswell <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Kellen Cresswell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
