<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Input data formats • TADCompare</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Input data formats">
<meta property="og:description" content="TADCompare">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TADCompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/TADCompare.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Input_Data.html">Input data formats</a>
    </li>
    <li>
      <a href="../articles/Ontology_Analysis.html">Gene Ontology Enrichment Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dozmorovlab/TADCompare/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Input_Data_files/header-attrs-2.5/header-attrs.js"></script><link href="Input_Data_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="Input_Data_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Input data formats</h1>
                        <h4 class="author">Kellen Cresswell</h4>
            <address class="author_afil">
      Department of Biostatistics, Virginia Commonwealth University, Richmond, VA<br><h4 class="author">Mikhail Dozmorov</h4>
            <address class="author_afil">
      Department of Biostatistics, Virginia Commonwealth University, Richmond, VA<br><small class="dont-index">Source: <a href="https://github.com/dozmorovlab/TADCompare/blob/master/vignettes/Input_Data.Rmd"><code>vignettes/Input_Data.Rmd</code></a></small>
      <div class="hidden name"><code>Input_Data.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>TADCompare is an R package for differential analysis of TAD boundaries. It is designed to work on a wide range of formats and resolutions of Hi-C data. TADCompare package contains four functions: <code>TADCompare</code>, <code>TimeCompare</code>, <code>ConsensusTADs</code>, and <code>DiffPlot</code>. <code>TADCompare</code> function allows for the identification of differential TAD boundaries between <a href="TADCompare.html#tadcompare">two contact matrices</a>. <code>TimeCompare</code> function takes a set of contact matrices, one matrix per time point, identifies TAD boundaries, and classifies <a href="TADCompare.html#timecompare">how they change over time</a>. <code>ConsensusTADs</code> function takes a list of TADs and identifies a consensus of TAD boundaries across all matrices using our <a href="TADCompare.html#consensustads">novel consensus boundary score</a>. <code>DiffPlot</code> allows for <a href="TADCompare.html#visualization">visualization of TAD boundary differences</a> between two matrices. The required input includes matrices in sparse 3-column format, <span class="math inline">\(n \times n\)</span>, or <span class="math inline">\(n \times (n+3)\)</span> formats. This vignette provides a complete overview of input data formats.</p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting Started</h1>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"TADCompare"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SpectralTAD</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dozmorovlab/TADCompare">TADCompare</a></span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="working-with-different-types-of-data" class="section level1">
<h1 class="hasAnchor">
<a href="#working-with-different-types-of-data" class="anchor"></a>Working with different types of data</h1>
<div id="working-with-n-times-n-matrices" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-n-times-n-matrices" class="anchor"></a>Working with <span class="math inline">\(n \times n\)</span> matrices</h2>
<p><span class="math inline">\(n \times n\)</span> contact matrices are most commonly associated with data coming from the Bing Ren lab (<a href="http://chromosome.sdsc.edu/mouse/hi-c/download.html" class="uri">http://chromosome.sdsc.edu/mouse/hi-c/download.html</a>). These contact matrices are square and symmetric with entry <span class="math inline">\(ij\)</span> corresponding to the number of contacts between region <span class="math inline">\(i\)</span> and region <span class="math inline">\(j\)</span>. Below is an example of a <span class="math inline">\(5 \times 5\)</span> region of an <span class="math inline">\(n \times n\)</span> contact matrix derived from Rao et al. 2014 data, GM12878 cell line <span class="citation">(Rao et al. 2014)</span>, chromosome 22, 50kb resolution. Note the symmetry around the diagonal - the typical shape of chromatin interaction matrix. The figure was created using the <a href="https://cran.r-project.org/web/packages/pheatmap/index.html">pheatmap</a> package.</p>
<p><img src="Input_Data_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
</div>
<div id="working-with-n-times-n3-matrices" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-n-times-n3-matrices" class="anchor"></a>Working with <span class="math inline">\(n \times (n+3)\)</span> matrices</h2>
<p><span class="math inline">\(n \times (n+3)\)</span> matrices are commonly associated with the <code>TopDom</code> TAD caller (<a href="http://zhoulab.usc.edu/TopDom/" class="uri">http://zhoulab.usc.edu/TopDom/</a>). These matrices consist of an <span class="math inline">\(n \times n\)</span> matrix but with three additional leading columns containing the chromosome, the start of the region and the end of the region. Regions in this case are determined by the resolution of the data. The subset of a typical <span class="math inline">\(n \times (n+3)\)</span> matrix is shown below.</p>
<pre><code>##     chr    start      end X18500000 X18550000 X18600000 X18650000
## 1 chr22 18500000 18550000     13313      4817      1664        96
## 2 chr22 18550000 18600000      4817     15500      5120       178
## 3 chr22 18600000 18650000      1664      5120     11242       316
## 4 chr22 18650000 18700000        96       178       316       162</code></pre>
</div>
<div id="working-with-sparse-3-column-matrices" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-sparse-3-column-matrices" class="anchor"></a>Working with sparse 3-column matrices</h2>
<p>Sparse 3-column matrices are matrices where the first and second columns refer to region <span class="math inline">\(i\)</span> and region <span class="math inline">\(j\)</span> of the chromosome, and the third column is the number of contacts between them. This style is becoming increasingly popular and is associated with raw data from Lieberman-Aiden lab (e.g., <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63525" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63525</a>), and is the data output produced by the Juicer tool <span class="citation">(Durand et al. 2016)</span>. 3-column matrices are handled internally in the package by converting them to <span class="math inline">\(n \times n\)</span> matrices using the <a href="https://bioconductor.org/packages/release/bioc/html/HiCcompare.html">HiCcompare</a> package’s <code>sparse2full()</code> function. The first 5 rows of a typical sparse 3-column matrix are shown below.</p>
<pre><code>##     region1  region2 IF
## 1: 16050000 16050000 12
## 2: 16200000 16200000  4
## 3: 16150000 16300000  1
## 4: 16200000 16300000  1
## 5: 16250000 16300000  1
## 6: 16300000 16300000 10</code></pre>
</div>
<div id="working-with-other-data-types" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-other-data-types" class="anchor"></a>Working with other data types</h2>
</div>
<div id="working-with--hic-files" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with--hic-files" class="anchor"></a>Working with .hic files</h2>
<p>.hic files are a common form of files generally associated with the lab of Erez Lieberman-Aiden (<a href="http://aidenlab.org/data.html" class="uri">http://aidenlab.org/data.html</a>). To use .hic files you must use the following steps.</p>
<ol style="list-style-type: decimal">
<li>Download <code>straw</code> from <a href="https://github.com/aidenlab/straw/" class="uri">https://github.com/aidenlab/straw/</a> and follow instalation instructions.</li>
<li>Download .hic data files. Here, we use data from Rao 2014 and download them using the following commands:</li>
</ol>
<p><code>wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63525/suppl/GSE63525_GM12878_insitu_primary_30.hic</code></p>
<p><code>wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63525/suppl/GSE63525_GM12878_insitu_replicate.hic</code></p>
<ol start="3" style="list-style-type: decimal">
<li>Extract chromosome 22 at 50kb resolution with no normalization:</li>
</ol>
<p><code>./straw NONE GSE63525_GM12878_insitu_primary_30.hic  22 22 BP 50000 &gt; primary.chr22.50kb.txt</code></p>
<p><code>./straw NONE GSE63525_GM12878_insitu_replicate_30.hic  22 22 BP 50000 &gt; replicate.chr22.50kb.txt</code></p>
<ol start="4" style="list-style-type: decimal">
<li>Analyze normally:</li>
</ol>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Read in data</span>
<span class="va">primary</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">'primary.chr22.50kb.txt'</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">replicate</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">'replicate.chr22.50kb.txt'</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#Run TADCompare</span>
<span class="va">tad_diff</span><span class="op">=</span><span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">primary</span>, <span class="va">replicate</span>, resolution<span class="op">=</span><span class="fl">50000</span><span class="op">)</span></code></pre></div>
</div>
<div id="working-with--cool-files" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with--cool-files" class="anchor"></a>Working with .cool files</h2>
<p>Users can also find TADs from data output by <code>cooler</code> (<a href="http://cooler.readthedocs.io/en/latest/index.html" class="uri">http://cooler.readthedocs.io/en/latest/index.html</a>) and HiC-Pro (<a href="https://github.com/nservant/HiC-Pro" class="uri">https://github.com/nservant/HiC-Pro</a>) with minor pre-processing using the <a href="https://bioconductor.org/packages/release/bioc/html/HiCcompare.html">HiCcompare</a> package.</p>
<p>The cooler software can be downloaded from <a href="https://mirnylab.github.io/cooler/" class="uri">https://mirnylab.github.io/cooler/</a>. A catalog of popular HiC datasets can be found at <a href="ftp://cooler.csail.mit.edu/coolers" class="uri">ftp://cooler.csail.mit.edu/coolers</a>. We can extract chromatin interaction data from .cool files using the following steps:</p>
<ol style="list-style-type: decimal">
<li>Follow instructions to install the cooler software, <a href="https://mirnylab.github.io/cooler/" class="uri">https://mirnylab.github.io/cooler/</a>
</li>
<li>Download the first contact matrix wget <a href="ftp://cooler.csail.mit.edu/coolers/hg19/Zuin2014-HEK293CtcfControl-HindIII-allreps-filtered.50kb.cool" class="uri">ftp://cooler.csail.mit.edu/coolers/hg19/Zuin2014-HEK293CtcfControl-HindIII-allreps-filtered.50kb.cool</a>
</li>
<li>Convert the first matrix to a text file using <code>cooler dump --join Zuin2014-HEK293CtcfControl-HindIII-allreps-filtered.50kb.cool &gt; Zuin.HEK293.50kb.Control.txt</code>
</li>
<li>Download the second contact matrix wget <a href="ftp://cooler.csail.mit.edu/coolers/hg19/Zuin2014-HEK293CtcfDepleted-HindIII-allreps-filtered.50kb.cool" class="uri">ftp://cooler.csail.mit.edu/coolers/hg19/Zuin2014-HEK293CtcfDepleted-HindIII-allreps-filtered.50kb.cool</a>
</li>
<li>Convert the matrix to a text file using <code>cooler dump --join Zuin2014-HEK293CtcfDepleted-HindIII-allreps-filtered.50kb.cool &gt; Zuin.HEK293.50kb.Depleted.txt</code>
</li>
<li>Run the code below</li>
</ol>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read in data</span>
<span class="va">cool_mat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"Zuin.HEK293.50kb.Control.txt"</span><span class="op">)</span>
<span class="va">cool_mat2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"Zuin.HEK293.50kb.Depleted.txt"</span><span class="op">)</span>

<span class="co"># Convert to sparse 3-column matrix using cooler2sparse from HiCcompare</span>
<span class="va">sparse_mat1</span> <span class="op">&lt;-</span> <span class="fu">HiCcompare</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiCcompare/man/cooler2sparse.html">cooler2sparse</a></span><span class="op">(</span><span class="va">cool_mat1</span><span class="op">)</span>
<span class="va">sparse_mat2</span> <span class="op">&lt;-</span> <span class="fu">HiCcompare</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiCcompare/man/cooler2sparse.html">cooler2sparse</a></span><span class="op">(</span><span class="va">cool_mat2</span><span class="op">)</span>

<span class="co"># Run TADCompare</span>
<span class="va">diff_tads</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">sparse_mat1</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">sparse_mat1</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span>, <span class="va">sparse_mat2</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span></code></pre></div>
</div>
<div id="working-with-hic-pro-files" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-hic-pro-files" class="anchor"></a>Working with HiC-Pro files</h2>
<p>HiC-Pro data is represented as two files, the <code>.matrix</code> file and the <code>.bed</code> file. The <code>.bed</code> file contains four columns (chromosome, start, end, ID). The <code>.matrix</code> file is a three-column matrix where the 1<sup>st</sup> and 2<sup>nd</sup> columns contain region IDs that map back to the coordinates in the bed file, and the third column contains the number of contacts between the two regions. In this example we analyze two matrix files <code>sample1_100000.matrix</code> and <code>sample2_100000.matrix</code>and their corresponding bed files <code>sample1_100000_abs.bed</code> and <code>sample2_100000_abs.bed</code>. We do not include HiC-Pro data in the package, so these serve as placeholders for the traditional files output by HiC-Pro. The steps for analyzing these files is shown below:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read in both files</span>
<span class="va">mat1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"sample1_100000.matrix"</span><span class="op">)</span>
<span class="va">bed1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"sample1_100000_abs.bed"</span><span class="op">)</span>

<span class="co"># Matrix 2</span>

<span class="va">mat2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"sample2_100000.matrix"</span><span class="op">)</span>
<span class="va">bed2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"sample2_100000_abs.bed"</span><span class="op">)</span>

<span class="co"># Convert to modified bed format</span>
<span class="va">sparse_mats1</span> <span class="op">&lt;-</span> <span class="fu">HiCcompare</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiCcompare/man/hicpro2bedpe.html">hicpro2bedpe</a></span><span class="op">(</span><span class="va">mat1</span>,<span class="va">bed1</span><span class="op">)</span>
<span class="va">sparse_mats2</span> <span class="op">&lt;-</span> <span class="fu">HiCcompare</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiCcompare/man/hicpro2bedpe.html">hicpro2bedpe</a></span><span class="op">(</span><span class="va">mat2</span>,<span class="va">bed2</span><span class="op">)</span>

<span class="co"># Remove empty matrices if necessary</span>
<span class="co"># sparse_mats$cis = sparse_mats$cis[sapply(sparse_mats, nrow) != 0]</span>


<span class="co"># Go through all pairwise chromosomes and run TADCompare</span>
<span class="va">sparse_tads</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">sparse_mats1</span><span class="op">$</span><span class="va">cis</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">z</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">sparse_mats1</span><span class="op">$</span><span class="va">cis</span><span class="op">[[</span><span class="va">z</span><span class="op">]</span><span class="op">]</span>
  <span class="va">y</span> <span class="op">&lt;-</span> <span class="va">sparse_mats2</span><span class="op">$</span><span class="va">cis</span><span class="op">[[</span><span class="va">z</span><span class="op">]</span><span class="op">]</span>
  
  <span class="co">#Pull out chromosome</span>
  <span class="va">chr</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
  <span class="co">#Subset to make three column matrix</span>
  <span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">5</span>, <span class="fl">7</span><span class="op">)</span><span class="op">]</span>
  <span class="va">y</span> <span class="op">&lt;-</span> <span class="va">y</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">5</span>, <span class="fl">7</span><span class="op">)</span><span class="op">]</span>
  <span class="co">#Run SpectralTAD</span>
  <span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, resolution <span class="op">=</span> <span class="fl">100000</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">comp</span>, <span class="va">chr</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Pull out differential TAD results</span>
<span class="va">diff_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">sparse_tads</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">$</span><span class="va">comp</span><span class="op">)</span>
<span class="co"># Pull out chromosomes</span>
<span class="va">chr</span>      <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">sparse_tads</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">$</span><span class="va">chr</span><span class="op">)</span>
<span class="co"># Name list by corresponding chr</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">diff_res</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">chr</span></code></pre></div>
</div>
<div id="effect-of-matrix-type-on-runtime" class="section level2">
<h2 class="hasAnchor">
<a href="#effect-of-matrix-type-on-runtime" class="anchor"></a>Effect of matrix type on runtime</h2>
<p>The type of matrix input into the algorithm can affect runtimes for the algorithm. <span class="math inline">\(n \times n\)</span> matrices require no conversion and are the fastest. Meanwhile, <span class="math inline">\(n \times (n+3)\)</span> matrices take slightly longer to run due to the need to remove the first 3 columns. Sparse 3-column matrices have the highest runtimes due to the complexity of converting them to an <span class="math inline">\(n \times n\)</span> matrix. The times are summarized below, holding all other parameters constant.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joshuaulrich/microbenchmark/">microbenchmark</a></span><span class="op">)</span>
<span class="co"># Reading in the second matrix</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"rao_chr22_rep"</span><span class="op">)</span>
<span class="co"># Converting to sparse</span>
<span class="va">prim_sparse</span> <span class="op">&lt;-</span> <span class="fu">HiCcompare</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiCcompare/man/full2sparse.html">full2sparse</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span><span class="op">)</span>
<span class="va">rep_sparse</span>  <span class="op">&lt;-</span> <span class="fu">HiCcompare</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HiCcompare/man/full2sparse.html">full2sparse</a></span><span class="op">(</span><span class="va">rao_chr22_rep</span><span class="op">)</span>
<span class="co"># Converting to nxn+3</span>
<span class="co"># Primary</span>
<span class="va">prim_n_n_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">"chr22"</span>,
                         start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span><span class="op">)</span><span class="op">)</span>,
                         end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span><span class="op">)</span><span class="op">)</span><span class="op">+</span><span class="fl">50000</span>, 
                         <span class="va">rao_chr22_prim</span><span class="op">)</span>

<span class="co"># Replicate</span>
<span class="va">rep_n_n_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">"chr22"</span>, 
                        start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">rao_chr22_rep</span><span class="op">)</span><span class="op">)</span>,
                        end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">rao_chr22_rep</span><span class="op">)</span><span class="op">)</span><span class="op">+</span><span class="fl">50000</span>,
                        <span class="va">rao_chr22_rep</span><span class="op">)</span>
<span class="co"># Defining each function</span>
<span class="co"># Sparse</span>
<span class="va">sparse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span>cont_mat1 <span class="op">=</span> <span class="va">prim_sparse</span>, cont_mat2 <span class="op">=</span> <span class="va">rep_sparse</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>
<span class="co"># NxN</span>
<span class="va">n_by_n</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span>cont_mat1 <span class="op">=</span> <span class="va">prim_sparse</span>, cont_mat2 <span class="op">=</span> <span class="va">rep_sparse</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>
<span class="co"># Nx(N+3)</span>
<span class="va">n_by_n_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span>cont_mat1 <span class="op">=</span> <span class="va">prim_n_n_3</span>, cont_mat2 <span class="op">=</span> <span class="va">rep_n_n_3</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>

<span class="co"># Benchmarking different parameters</span>
<span class="va">bench</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html">microbenchmark</a></span><span class="op">(</span>
<span class="co"># Sparse</span>
<span class="va">sparse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span>cont_mat1 <span class="op">=</span> <span class="va">prim_sparse</span>, cont_mat2 <span class="op">=</span> <span class="va">rep_sparse</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>,
<span class="co"># NxN</span>
<span class="va">n_by_n</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span>cont_mat1 <span class="op">=</span> <span class="va">rao_chr22_prim</span>, cont_mat2 <span class="op">=</span> <span class="va">rao_chr22_rep</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>,
<span class="co"># Nx(N+3)</span>
<span class="va">n_by_n_3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span>cont_mat1 <span class="op">=</span> <span class="va">prim_n_n_3</span>, cont_mat2 <span class="op">=</span> <span class="va">rep_n_n_3</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>, times <span class="op">=</span> <span class="fl">5</span>, unit <span class="op">=</span> <span class="st">"s"</span>
<span class="op">)</span> 

<span class="va">summary_bench</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">bench</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">mean</span>, <span class="va">median</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">summary_bench</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sparse"</span>, <span class="st">"n_by_n"</span>, <span class="st">"n_by_n_3"</span><span class="op">)</span>
<span class="va">summary_bench</span></code></pre></div>
<pre><code>##                mean     median
## sparse   0.24663187 0.16743460
## n_by_n   0.08653424 0.09028903
## n_by_n_3 0.09875885 0.10077293</code></pre>
<p>The table above shows the mean and median of runtimes for different types of contact matrices measured in seconds. As we see, <code>TADCompare</code> is extremely fast irrespectively of the parameters. However, sparse matrix inputs will slow down the algorithm. This can become more apparent as the size of the contact matrices increase.</p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS  10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] microbenchmark_1.4-7 TADCompare_1.1.1     SpectralTAD_1.4.0   
## [4] dplyr_1.0.2          BiocStyle_2.16.1    
## 
## loaded via a namespace (and not attached):
##   [1] TH.data_1.0-10              colorspace_2.0-0           
##   [3] ggsignif_0.6.0              ellipsis_0.3.1             
##   [5] rio_0.5.16                  CGHcall_2.50.0             
##   [7] rprojroot_2.0.2             DNAcopy_1.62.0             
##   [9] XVector_0.28.0              GenomicRanges_1.40.0       
##  [11] fs_1.5.0                    rstudioapi_0.13            
##  [13] ggpubr_0.4.0                listenv_0.8.0              
##  [15] mvtnorm_1.1-1               HiCcompare_1.10.0          
##  [17] codetools_0.2-16            splines_4.0.2              
##  [19] R.methodsS3_1.8.1           impute_1.62.0              
##  [21] knitr_1.30                  Rsamtools_2.4.0            
##  [23] broom_0.7.2                 R.oo_1.24.0                
##  [25] pheatmap_1.0.12             BiocManager_1.30.10        
##  [27] compiler_4.0.2              backports_1.2.0            
##  [29] assertthat_0.2.1            Matrix_1.2-18              
##  [31] limma_3.44.3                htmltools_0.5.0            
##  [33] tools_4.0.2                 gtable_0.3.0               
##  [35] glue_1.4.2                  GenomeInfoDbData_1.2.3     
##  [37] reshape2_1.4.4              Rcpp_1.0.5                 
##  [39] carData_3.0-4               Biobase_2.48.0             
##  [41] cellranger_1.1.0            pkgdown_1.6.1              
##  [43] vctrs_0.3.5                 Biostrings_2.56.0          
##  [45] nlme_3.1-150                QDNAseq_1.24.0             
##  [47] xfun_0.19                   stringr_1.4.0              
##  [49] globals_0.13.1              openxlsx_4.2.3             
##  [51] lifecycle_0.2.0             gtools_3.8.2               
##  [53] rstatix_0.6.0               InteractionSet_1.16.0      
##  [55] future_1.20.1               MASS_7.3-53                
##  [57] zoo_1.8-8                   zlibbioc_1.34.0            
##  [59] scales_1.1.1                hms_0.5.3                  
##  [61] ragg_0.4.0                  sandwich_3.0-0             
##  [63] parallel_4.0.2              SummarizedExperiment_1.18.2
##  [65] rhdf5_2.32.4                RColorBrewer_1.1-2         
##  [67] yaml_2.2.1                  curl_4.3                   
##  [69] memoise_1.1.0               gridExtra_2.3              
##  [71] ggplot2_3.3.2               CGHbase_1.48.0             
##  [73] stringi_1.5.3               S4Vectors_0.26.1           
##  [75] desc_1.2.0                  BiocGenerics_0.34.0        
##  [77] zip_2.1.1                   BiocParallel_1.22.0        
##  [79] GenomeInfoDb_1.24.2         rlang_0.4.8                
##  [81] pkgconfig_2.0.3             systemfonts_0.3.2          
##  [83] matrixStats_0.57.0          bitops_1.0-6               
##  [85] evaluate_0.14               lattice_0.20-41            
##  [87] purrr_0.3.4                 Rhdf5lib_1.10.1            
##  [89] cowplot_1.1.0               tidyselect_1.1.0           
##  [91] parallelly_1.21.0           plyr_1.8.6                 
##  [93] magrittr_2.0.1              bookdown_0.21              
##  [95] R6_2.5.0                    IRanges_2.22.2             
##  [97] generics_0.1.0              multcomp_1.4-15            
##  [99] DelayedArray_0.14.1         pillar_1.4.7               
## [101] haven_2.3.1                 foreign_0.8-80             
## [103] mgcv_1.8-33                 survival_3.2-7             
## [105] abind_1.4-5                 RCurl_1.98-1.2             
## [107] tibble_3.0.4                future.apply_1.6.0         
## [109] PRIMME_3.1-3                crayon_1.3.4               
## [111] car_3.0-10                  KernSmooth_2.23-18         
## [113] rmarkdown_2.5               readxl_1.3.1               
## [115] grid_4.0.2                  data.table_1.13.2          
## [117] marray_1.66.0               forcats_0.5.0              
## [119] digest_0.6.27               tidyr_1.1.2                
## [121] R.utils_2.10.1              textshaping_0.2.1          
## [123] stats4_4.0.2                munsell_0.5.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Durand:2016aa" class="csl-entry">
Durand, Neva C., Muhammad S. Shamim, Ido Machol, Suhas S. P. Rao, Miriam H. Huntley, Eric S. Lander, and Erez Lieberman Aiden. 2016. <span>“Juicer Provides a One-Click System for Analyzing Loop-Resolution Hi-c Experiments.”</span> <em>Cell Systems</em> 3 (1): 95–98. <a href="https://doi.org/10.1016/j.cels.2016.07.002">https://doi.org/10.1016/j.cels.2016.07.002</a>.
</div>
<div id="ref-Rao:2014aa" class="csl-entry">
Rao, Suhas S.P., Miriam H. Huntley, Neva C. Durand, Elena K. Stamenova, Ivan D. Bochkov, James T. Robinson, Adrian L. Sanborn, et al. 2014. <span>“A 3d Map of the Human Genome at Kilobase Resolution Reveals Principles of Chromatin Looping.”</span> <em>Cell</em> 159 (7): 1665–80. <a href="https://doi.org/10.1016/j.cell.2014.11.021">https://doi.org/10.1016/j.cell.2014.11.021</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kellen Cresswell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
