<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Gene Ontology Enrichment Analysis • TADCompare</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Gene Ontology Enrichment Analysis">
<meta property="og:description" content="TADCompare">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TADCompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/TADCompare.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Input_Data.html">Input data formats</a>
    </li>
    <li>
      <a href="../articles/Ontology_Analysis.html">Gene Ontology Enrichment Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dozmorovlab/TADCompare/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Ontology_Analysis_files/header-attrs-2.5/header-attrs.js"></script><link href="Ontology_Analysis_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="Ontology_Analysis_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Gene Ontology Enrichment Analysis</h1>
                        <h4 class="author">Kellen Cresswell</h4>
            <address class="author_afil">
      Department of Biostatistics, Virginia Commonwealth University, Richmond, VA<br><h4 class="author">Mikhail Dozmorov</h4>
            <address class="author_afil">
      Department of Biostatistics, Virginia Commonwealth University, Richmond, VA<br><small class="dont-index">Source: <a href="https://github.com/dozmorovlab/TADCompare/blob/master/vignettes/Ontology_Analysis.Rmd"><code>vignettes/Ontology_Analysis.Rmd</code></a></small>
      <div class="hidden name"><code>Ontology_Analysis.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"TADCompare"</span><span class="op">)</span></code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Using the output of <code>TADCompare</code> and <code>TimeCompare</code>, we can do a range of analyses. One common one is gene ontology enrichment analysis to determine the pathways in which genes near TAD boundaries occur in. To do this, we use <a href="https://bioconductor.org/packages/release/bioc/html/rGREAT.html">rGREAT</a> an R package for performing gene ontology enrichment analysis.</p>
</div>
<div id="performing-gene-ontology-analysis-using-tadcompare" class="section level1">
<h1 class="hasAnchor">
<a href="#performing-gene-ontology-analysis-using-tadcompare" class="anchor"></a>Performing gene ontology analysis using TADCompare</h1>
<p>In the first example, we show how to perform gene ontology enrichment using differential boundaries. Here, we perform the analysis on shifted boundaries detected in matrix 1.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/rGREAT">rGREAT</a></span><span class="op">)</span>
<span class="co"># Reading in data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"rao_chr22_prim"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"rao_chr22_rep"</span><span class="op">)</span>

<span class="co"># Performing differential analysis</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span>, <span class="va">rao_chr22_rep</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>

<span class="co"># Saving the results into its own data frame</span>
<span class="va">TAD_Frame</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">TAD_Frame</span>

<span class="co"># Filter data to only include complex boundaries enriched in the second</span>
<span class="co"># contact matrix</span>
<span class="va">TAD_Frame</span> <span class="op">&lt;-</span> <span class="va">TAD_Frame</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">(</span><span class="va">Type</span> <span class="op">==</span> <span class="st">"Shifted"</span><span class="op">)</span> <span class="op">&amp;</span> 
                                         <span class="op">(</span><span class="va">Enriched_In</span> <span class="op">==</span> <span class="st">"Matrix 2"</span><span class="op">)</span><span class="op">)</span>

<span class="co"># Assign a chromosome and convert to a bed format</span>
<span class="va">TAD_Frame</span> <span class="op">&lt;-</span> <span class="va">TAD_Frame</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">Boundary</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">"chr22"</span>, 
    start <span class="op">=</span> <span class="va">Boundary</span>, end <span class="op">=</span> <span class="va">Boundary</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span><span class="op">)</span>

<span class="co"># Set up rGREAT job with default parameters</span>
<span class="va">great_shift</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/submitGreatJob.html">submitGreatJob</a></span><span class="op">(</span><span class="va">TAD_Frame</span>, request_interval <span class="op">=</span> <span class="fl">1</span>, version <span class="op">=</span> <span class="st">"2.0"</span><span class="op">)</span>

<span class="co"># Submit the job</span>
<span class="va">enrichment_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getEnrichmentTables-GreatJob-method.html">getEnrichmentTables</a></span><span class="op">(</span><span class="va">great_shift</span><span class="op">)</span>

<span class="co"># Subset to only include vital information</span>
<span class="va">enrichment_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">enrichment_table</span>, .id <span class="op">=</span> <span class="st">"source"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>Ontology <span class="op">=</span> <span class="va">source</span>, Description <span class="op">=</span> <span class="va">name</span>, 
                `P-value` <span class="op">=</span> <span class="va">Hyper_Raw_PValue</span><span class="op">)</span>

<span class="co"># Print head organizaed by p-values</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">enrichment_table</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">`P-value`</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##                Ontology                                          Description
## 1 GO Molecular Function                   gamma-glutamyltransferase activity
## 2 GO Biological Process                     glutathione biosynthetic process
## 3 GO Cellular Component         anchored to external side of plasma membrane
## 4 GO Cellular Component        intrinsic to external side of plasma membrane
## 5 GO Biological Process                         peptide biosynthetic process
## 6 GO Molecular Function transferase activity, transferring amino-acyl groups
##       P-value
## 1 0.001802360
## 2 0.002927596
## 3 0.003152529
## 4 0.004051881
## 5 0.004725996
## 6 0.004950625</code></pre>
<p>The first column, “Ontology,” is simply the domain from which the corresponding ontology (“Description” column) comes from. Here, we use the default, which is the GO ontologies. For more available ontologies, see the <a href="https://bioconductor.org/packages/release/bioc/vignettes/rGREAT/inst/doc/rGREAT.html">rGREAT vignette</a>. “Description” is the pathway itself. “P-value” is the unadjusted hypergeometric p-value, as output by <code>rGREAT</code>. <code>rGREAT</code> also provides binomial p-values (Binom_Raw_Pvalue, Binom_Adjp_BH) and adjusted hypergeometric p-values (Hyper_Adjp_BH).</p>
<p>Now we demonstrate how to perform the same analysis but for all boundary types simultaneously. In this case, we use time-varying data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Read in time course data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"time_mats"</span><span class="op">)</span>
<span class="co"># Identifying boundaries</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TimeCompare.html">TimeCompare</a></span><span class="op">(</span><span class="va">time_mats</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>

<span class="co"># Pulling out the frame of TADs</span>
<span class="va">TAD_Frame</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">TAD_Bounds</span>

<span class="co"># Getting coordinates for TAD boundaries and converting into bed format</span>
<span class="va">Bound_List</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">TAD_Frame</span><span class="op">$</span><span class="va">Category</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">TAD_Frame</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">(</span><span class="va">Category</span> <span class="op">==</span> <span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>chr <span class="op">=</span> <span class="st">"chr22"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">chr</span>, <span class="va">Coordinate</span><span class="op">)</span> <span class="op">%&gt;%</span> 
        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">Coordinate</span>, end <span class="op">=</span> <span class="va">Coordinate</span><span class="op">)</span> <span class="op">%&gt;%</span> 
        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">chr</span>, <span class="va">start</span>, <span class="va">end</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Performing rGREAT analysis for each boundary Category</span>
<span class="va">TAD_Enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">Bound_List</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/getEnrichmentTables-GreatJob-method.html">getEnrichmentTables</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/rGREAT/man/submitGreatJob.html">submitGreatJob</a></span><span class="op">(</span><span class="va">x</span>, request_interval <span class="op">=</span> <span class="fl">1</span>, version <span class="op">=</span> <span class="st">"2.0"</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Name list of data frames to keep track of which enrichment belongs to which</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TAD_Enrich</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">TAD_Frame</span><span class="op">$</span><span class="va">Category</span><span class="op">)</span>

<span class="co"># Bind each category of pathway and create new column for each pathway</span>
<span class="va">TAD_Enrich</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">TAD_Enrich</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">TAD_Enrich</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span>, <span class="kw">function</span><span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">y</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Category <span class="op">=</span> <span class="va">x</span><span class="op">)</span>
  <span class="op">}</span><span class="op">)</span>, .id <span class="op">=</span> <span class="st">"source"</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="co"># Bind each boundary category together and pull out important variables</span>
<span class="va">enrichment_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">TAD_Enrich</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>Ontology <span class="op">=</span> <span class="va">source</span>, Description <span class="op">=</span> <span class="va">name</span>, 
                `P-value` <span class="op">=</span> <span class="va">Hyper_Raw_PValue</span>, <span class="va">Category</span><span class="op">)</span>

<span class="co"># Get the top enriched pathways</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">enrichment_table</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">`P-value`</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##                Ontology
## 1 GO Biological Process
## 2 GO Molecular Function
## 3 GO Biological Process
## 4 GO Biological Process
## 5 GO Biological Process
## 6 GO Molecular Function
##                                                Description      P-value
## 1 positive regulation of B cell receptor signaling pathway 0.0002254283
## 2                               lipid transporter activity 0.0003760684
## 3          regulation of B cell receptor signaling pathway 0.0004508185
## 4                               Schwann cell proliferation 0.0006761897
## 5                            lipoprotein metabolic process 0.0007139088
## 6        peptide-methionine (R)-S-oxide reductase activity 0.0009015354
##              Category
## 1         Dynamic TAD
## 2   Highly Common TAD
## 3         Dynamic TAD
## 4 Early Appearing TAD
## 5   Highly Common TAD
## 6   Highly Common TAD</code></pre>
<p>These columns are the same as the differential analysis but with an extra column, “Category,” indicating the type of time-varying TAD boundary.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS  10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] rGREAT_1.20.0        GenomicRanges_1.40.0 GenomeInfoDb_1.24.2 
## [4] IRanges_2.22.2       S4Vectors_0.26.1     BiocGenerics_0.34.0 
## [7] TADCompare_1.1.1     dplyr_1.0.2          BiocStyle_2.16.1    
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-0            rjson_0.2.20               
##   [3] ggsignif_0.6.0              ellipsis_0.3.1             
##   [5] rio_0.5.16                  CGHcall_2.50.0             
##   [7] rprojroot_2.0.2             DNAcopy_1.62.0             
##   [9] XVector_0.28.0              GlobalOptions_0.1.2        
##  [11] fs_1.5.0                    rstudioapi_0.13            
##  [13] ggpubr_0.4.0                listenv_0.8.0              
##  [15] HiCcompare_1.10.0           codetools_0.2-16           
##  [17] splines_4.0.2               R.methodsS3_1.8.1          
##  [19] impute_1.62.0               knitr_1.30                 
##  [21] Rsamtools_2.4.0             broom_0.7.2                
##  [23] R.oo_1.24.0                 pheatmap_1.0.12            
##  [25] BiocManager_1.30.10         compiler_4.0.2             
##  [27] backports_1.2.0             assertthat_0.2.1           
##  [29] Matrix_1.2-18               limma_3.44.3               
##  [31] htmltools_0.5.0             tools_4.0.2                
##  [33] gtable_0.3.0                glue_1.4.2                 
##  [35] GenomeInfoDbData_1.2.3      reshape2_1.4.4             
##  [37] Rcpp_1.0.5                  carData_3.0-4              
##  [39] Biobase_2.48.0              cellranger_1.1.0           
##  [41] pkgdown_1.6.1               vctrs_0.3.5                
##  [43] Biostrings_2.56.0           nlme_3.1-150               
##  [45] QDNAseq_1.24.0              xfun_0.19                  
##  [47] stringr_1.4.0               globals_0.13.1             
##  [49] openxlsx_4.2.3              lifecycle_0.2.0            
##  [51] gtools_3.8.2                rstatix_0.6.0              
##  [53] InteractionSet_1.16.0       future_1.20.1              
##  [55] zlibbioc_1.34.0             scales_1.1.1               
##  [57] ragg_0.4.0                  hms_0.5.3                  
##  [59] SummarizedExperiment_1.18.2 rhdf5_2.32.4               
##  [61] RColorBrewer_1.1-2          yaml_2.2.1                 
##  [63] curl_4.3                    memoise_1.1.0              
##  [65] gridExtra_2.3               ggplot2_3.3.2              
##  [67] CGHbase_1.48.0              stringi_1.5.3              
##  [69] desc_1.2.0                  zip_2.1.1                  
##  [71] BiocParallel_1.22.0         rlang_0.4.8                
##  [73] pkgconfig_2.0.3             systemfonts_0.3.2          
##  [75] matrixStats_0.57.0          bitops_1.0-6               
##  [77] evaluate_0.14               lattice_0.20-41            
##  [79] purrr_0.3.4                 Rhdf5lib_1.10.1            
##  [81] cowplot_1.1.0               tidyselect_1.1.0           
##  [83] parallelly_1.21.0           plyr_1.8.6                 
##  [85] magrittr_2.0.1              bookdown_0.21              
##  [87] R6_2.5.0                    generics_0.1.0             
##  [89] DelayedArray_0.14.1         pillar_1.4.7               
##  [91] haven_2.3.1                 foreign_0.8-80             
##  [93] mgcv_1.8-33                 abind_1.4-5                
##  [95] RCurl_1.98-1.2              tibble_3.0.4               
##  [97] future.apply_1.6.0          PRIMME_3.1-3               
##  [99] crayon_1.3.4                car_3.0-10                 
## [101] KernSmooth_2.23-18          rmarkdown_2.5              
## [103] GetoptLong_1.0.4            grid_4.0.2                 
## [105] readxl_1.3.1                data.table_1.13.2          
## [107] marray_1.66.0               forcats_0.5.0              
## [109] digest_0.6.27               tidyr_1.1.2                
## [111] R.utils_2.10.1              textshaping_0.2.1          
## [113] munsell_0.5.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kellen Cresswell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
