<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>TAD comparison between two conditions • TADCompare</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="TAD comparison between two conditions">
<meta property="og:description" content="TADCompare">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TADCompare</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/TADCompare.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Input_Data.html">Input data formats</a>
    </li>
    <li>
      <a href="../articles/Ontology_Analysis.html">Gene Ontology Enrichment Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dozmorovlab/TADCompare/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="TADCompare_files/header-attrs-2.5/header-attrs.js"></script><link href="TADCompare_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="TADCompare_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>TAD comparison between two conditions</h1>
                        <h4 class="author">Kellen Cresswell</h4>
            <address class="author_afil">
      Department of Biostatistics, Virginia Commonwealth University, Richmond, VA<br><h4 class="author">Mikhail Dozmorov</h4>
            <address class="author_afil">
      Department of Biostatistics, Virginia Commonwealth University, Richmond, VA<br><small class="dont-index">Source: <a href="https://github.com/dozmorovlab/TADCompare/blob/master/vignettes/TADCompare.Rmd"><code>vignettes/TADCompare.Rmd</code></a></small>
      <div class="hidden name"><code>TADCompare.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"TADCompare"</span><span class="op">)</span>
<span class="co"># Or, for the developmental version</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"dozmorovlab/TADCompare"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SpectralTAD</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dozmorovlab/TADCompare">TADCompare</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="tadcompare" class="section level1">
<h1 class="hasAnchor">
<a href="#tadcompare" class="anchor"></a>TADCompare</h1>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p><code>TADCompare</code> is a function that allows users to automatically identify differential TAD boundaries between two datasets. For each differential boundary, we provide unique classification (complex, merge, split, shifted, and strength change) defining how the TAD boundaries change between the datasets.</p>
</div>
<div id="running-tadcompare" class="section level2">
<h2 class="hasAnchor">
<a href="#running-tadcompare" class="anchor"></a>Running TADcompare</h2>
<p>The only required input is two contact matrices in one of the permitted forms specified in the <a href="Input_Data.html">Input data vignette</a>. <code>TADCompare</code> function will automatically determine the type of matrix and convert it to an appropriate form, given it is one of the supported formats. The only requirement is that all matrices be in the same format. For the fastest results, we suggest using <span class="math inline">\(n \times n\)</span> matrices. Additionally, we recommend users to provide resolution of their data. If the resolution is not provided, we estimate it using the numeric column names of contact matrices. We illustrate running <code>TADCompare</code> using the data from GM12878 cell line, chromosome 22, 50kb resolution <span class="citation">(Suhas S. P. Rao et al. 2014)</span>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get the rao contact matrices built into the package</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"rao_chr22_prim"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"rao_chr22_rep"</span><span class="op">)</span>
<span class="co"># We see these are n x n matrices</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 704 704</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">rao_chr22_rep</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 704 704</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># And, get a glimpse how the data looks like</span>
<span class="va">rao_chr22_prim</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">105</span>, <span class="fl">100</span><span class="op">:</span><span class="fl">105</span><span class="op">]</span></code></pre></div>
<pre><code>##          2.1e+07 21050000 21100000 21150000 21200000 21250000
## 2.1e+07     6215     1047     1073      714      458      383
## 21050000    1047     4542     2640     1236      619      489
## 21100000    1073     2640    13468     4680     1893     1266
## 21150000     714     1236     4680    13600     4430     2124
## 21200000     458      619     1893     4430    11313     4465
## 21250000     383      489     1266     2124     4465    11824</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Running the algorithm with resolution specified</span>
<span class="va">results</span> <span class="op">=</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span>, <span class="va">rao_chr22_rep</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span>
<span class="co"># Repeating without specifying resolution</span>
<span class="va">no_res</span> <span class="op">=</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span>, <span class="va">rao_chr22_rep</span><span class="op">)</span></code></pre></div>
<pre><code>## Estimating resolution</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># We can see below that resolution can be estimated automatically if necessary</span>
<span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">Diff_Loci</span>, <span class="va">no_res</span><span class="op">$</span><span class="va">Diff_Loci</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="types-of-tadcompare-output" class="section level2">
<h2 class="hasAnchor">
<a href="#types-of-tadcompare-output" class="anchor"></a>Types of TADCompare output</h2>
<p><code>TADCompare</code> function returns a list with two data frames and one plot. The first data frame contains all regions of the genome containing a TAD boundary in at least one of the contact matrices. The results are shown below:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">TAD_Frame</span><span class="op">)</span></code></pre></div>
<pre><code>##   Boundary   Gap_Score TAD_Score1 TAD_Score2     Differential Enriched_In
## 1 16300000 -18.6589859  0.7698503   7.704411     Differential    Matrix 2
## 2 16850000   0.2361100  7.8431724   7.580056 Non-Differential    Matrix 1
## 3 17350000   0.9683905  3.6628411   3.220253 Non-Differential    Matrix 1
## 4 18000000  -0.1033894  2.3616107   2.347392 Non-Differential    Matrix 2
## 5 18750000   3.0426660  4.7997622   3.558975     Differential    Matrix 1
## 6 18850000   2.6911660  3.7674844   2.680707     Differential    Matrix 1
##               Type
## 1             &lt;NA&gt;
## 2 Non-Differential
## 3 Non-Differential
## 4 Non-Differential
## 5  Strength Change
## 6  Strength Change</code></pre>
<p>The “Boundary” column contains genomic coordinates of the given boundary. “Gap_Score” corresponds to the differential boundary score (Z-score of the difference between boundary scores). “TAD_Score1” and “TAD_Score2” corresponds to the boundary score of the two contact matrices. “Differential” simply indicates whether the boundary is differential or not differential. “Enriched_In” indicates which matrix contains the TAD boundary. “Type” identifies the type of TAD boundary change. Note: The first boundary will always be classified as “NA” since specific classification is impossible without a reference boundary to the left and right.</p>
<p>The second data frame contains the same information as the first data frame but includes every region of the genome. We show it below:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">Boundary_Scores</span><span class="op">)</span></code></pre></div>
<pre><code>##   Boundary TAD_Score1 TAD_Score2     Gap_Score     Differential Enriched_In
## 1 16300000  0.7698503  7.7044114 -18.658985912     Differential    Matrix 2
## 2 16850000  7.8431724  7.5800558   0.236110024 Non-Differential    Matrix 1
## 3 16900000 -0.4402121 -0.4337948   0.009161239 Non-Differential    Matrix 1
## 4 16950000 -0.7460531 -0.5551327  -0.467725582 Non-Differential    Matrix 2
## 5 17000000 -0.6803509 -0.6512025  -0.037456910 Non-Differential    Matrix 2
## 6 17050000 -0.6626352 -0.5562997  -0.245694062 Non-Differential    Matrix 2
##               Type
## 1             &lt;NA&gt;
## 2 Non-Differential
## 3 Non-Differential
## 4 Non-Differential
## 5 Non-Differential
## 6 Non-Differential</code></pre>
<p>Finally, we include a plot that contains a stacked barplot indicating the prevalence of each type of TAD boundary. The barplot is a <code>ggplot2</code> object, making it completely customizable. We show this below:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">Count_Plot</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "gg"     "ggplot"</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Removed 1 rows containing missing values (position_stack).</code></pre>
<p><img src="TADCompare_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div id="pre-specification-of-tads" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-specification-of-tads" class="anchor"></a>Pre-specification of TADs</h2>
<p>We recognize that users may like to use their own TAD caller for the identification of TAD boundaries prior to boundary comparison. As a result, we provide the option for pre-specification of TADs in a form of two data frames with chromosome, start, and end coordinates of TAD boundaries for the two contact matrices. With this option, only provided TAD boundaries will be tested. We provide an example below using the <a href="https://bioconductor.org/packages/release/bioc/html/SpectralTAD.html">SpectralTAD</a> TAD caller <span class="citation">(Cresswell, Stansfield, and Dozmorov 2019)</span>:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Call TADs using SpectralTAD</span>
<span class="va">bed_coords1</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="fu">SpectralTAD</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SpectralTAD/man/SpectralTAD.html">SpectralTAD</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span>, chr <span class="op">=</span> <span class="st">"chr22"</span>, levels <span class="op">=</span> <span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Estimating resolution</code></pre>
<pre><code>## Warning: Unknown or uninitialised column: `Group`.

## Warning: Unknown or uninitialised column: `Group`.</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bed_coords2</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpectralTAD/man/SpectralTAD.html">SpectralTAD</a></span><span class="op">(</span><span class="va">rao_chr22_rep</span>,  chr <span class="op">=</span> <span class="st">"chr22"</span>, levels <span class="op">=</span> <span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Estimating resolution</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Combining the TAD boundaries for both contact matrices</span>
<span class="va">Combined_Bed</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">bed_coords1</span>, <span class="va">bed_coords2</span><span class="op">)</span>

<span class="co"># Printing the combined list of TAD boundaries</span></code></pre></div>
<p>Above, we see the dataset output by SpectralTAD contains a column named “start” and “end” indicating the start and end coordinate of each TAD. This is required, though the output of any TAD caller can be used effectively with some data manipulation. The “Level” column indicates the level of TAD in a hierarchy.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Running TADCompare with pre-specified TADs</span>
<span class="va">TD_Compare</span> <span class="op">=</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">rao_chr22_prim</span>, <span class="va">rao_chr22_rep</span>, resolution <span class="op">=</span> <span class="fl">50000</span>, pre_tads <span class="op">=</span> <span class="va">Combined_Bed</span><span class="op">)</span>

<span class="co"># Returning the boundaries</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">TD_Compare</span><span class="op">$</span><span class="va">TAD_Frame</span><span class="op">)</span></code></pre></div>
<pre><code>##   Boundary   Gap_Score TAD_Score1 TAD_Score2     Differential Enriched_In
## 1 16850000  0.23611002  7.8431724  7.5800558 Non-Differential    Matrix 1
## 2 17250000 -0.08875306  0.1103987  0.1409999 Non-Differential    Matrix 2
## 3 17600000 -0.06733403 -0.4153272 -0.3809658 Non-Differential    Matrix 2
## 4 18000000 -0.10338944  2.3616107  2.3473922 Non-Differential    Matrix 2
## 5 18350000 -0.04284963 -0.7767081 -0.7433986 Non-Differential    Matrix 2
## 6 19100000  0.01857312 -0.8269469 -0.8153997 Non-Differential    Matrix 1
##               Type
## 1      Non-Overlap
## 2      Non-Overlap
## 3      Non-Overlap
## 4 Non-Differential
## 5 Non-Differential
## 6      Non-Overlap</code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Testing to show that all pre-specified boundaries are returned</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">TD_Compare</span><span class="op">$</span><span class="va">TAD_Frame</span><span class="op">$</span><span class="va">Boundary</span> <span class="op">%in%</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="va">Combined_Bed</span><span class="op">)</span><span class="op">$</span><span class="va">end</span><span class="op">)</span> </code></pre></div>
<pre><code>## 
## TRUE 
##   94</code></pre>
<p>Here, we see that the only boundaries that are returned are those we pre-specified.</p>
</div>
<div id="visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization of TADCompare Results</h2>
<p>We provide means for visualizing differential TAD boundaries using the <code>DiffPlot</code> function. As an input, <code>DiffPlot</code> takes the output from the <code>TADCompare</code> function and the original contact matrices. As an output, it returns a <code>ggplot2</code> object that can be further customized. We demonstrate visualization of the differences between GM12878 and IMR90 cell lines, on the subset of chromosome 2, 40kb resolution data processed in Schmitt et al. <span class="citation">(Schmitt et al. 2016)</span>:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"GM12878.40kb.raw.chr2"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"IMR90.40kb.raw.chr2"</span><span class="op">)</span>
<span class="va">mtx1</span> <span class="op">&lt;-</span> <span class="va">GM12878.40kb.raw.chr2</span>
<span class="va">mtx2</span> <span class="op">&lt;-</span> <span class="va">IMR90.40kb.raw.chr2</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fl">40000</span> <span class="co"># Set resolution</span>
<span class="co"># Globally normalize matrices for better visual comparison (does not affect TAD calling)</span>
<span class="va">mtx1_total</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">mtx1</span><span class="op">)</span>
<span class="va">mtx2_total</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">mtx2</span><span class="op">)</span>
<span class="op">(</span><span class="va">scaling_factor</span> <span class="op">&lt;-</span> <span class="va">mtx1_total</span> <span class="op">/</span> <span class="va">mtx2_total</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 0.3837505</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Rescale matrices depending on which matrix is smaller</span>
<span class="kw">if</span> <span class="op">(</span><span class="va">mtx1_total</span> <span class="op">&gt;</span> <span class="va">mtx2_total</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">mtx2</span> <span class="op">&lt;-</span> <span class="va">mtx2</span> <span class="op">*</span> <span class="va">scaling_factor</span>
<span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
  <span class="va">mtx1</span> <span class="op">&lt;-</span> <span class="va">mtx1</span> <span class="op">*</span> <span class="op">(</span><span class="fl">1</span> <span class="op">/</span> <span class="va">scaling_factor</span><span class="op">)</span>
<span class="op">}</span>
<span class="co"># Coordinates of interesting regions</span>
<span class="va">start_coord</span> <span class="op">&lt;-</span> <span class="fl">8000000</span>
<span class="va">end_coord</span>   <span class="op">&lt;-</span> <span class="fl">16000000</span>
<span class="co"># Another interesting region</span>
<span class="co"># start_coord &lt;- 40000000</span>
<span class="co"># end_coord   &lt;- 48000000</span></code></pre></div>
<div id="comparing-tad-boundary-scores" class="section level3">
<h3 class="hasAnchor">
<a href="#comparing-tad-boundary-scores" class="anchor"></a>Comparing TAD boundary scores</h3>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Running TADCompare as-is</span>
<span class="va">TD_Compare</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">mtx1</span>, <span class="va">mtx2</span>, resolution <span class="op">=</span> <span class="va">res</span><span class="op">)</span>
<span class="co"># Running the plotting algorithm with pre-specified TADs</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DiffPlot.html">DiffPlot</a></span><span class="op">(</span>tad_diff    <span class="op">=</span> <span class="va">TD_Compare</span>, 
              cont_mat1   <span class="op">=</span> <span class="va">mtx1</span>,
              cont_mat2   <span class="op">=</span> <span class="va">mtx2</span>,
              resolution  <span class="op">=</span> <span class="va">res</span>,
              start_coord <span class="op">=</span> <span class="va">start_coord</span>,
              end_coord   <span class="op">=</span> <span class="va">end_coord</span>,
              show_types  <span class="op">=</span> <span class="cn">TRUE</span>, 
              point_size  <span class="op">=</span> <span class="fl">5</span>,
              max_height  <span class="op">=</span> <span class="fl">5</span>,
              rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,
              palette     <span class="op">=</span> <span class="st">"RdYlBu"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<p><img src="TADCompare_files/figure-html/unnamed-chunk-10-1.png" width="960"></p>
<p>From these results, we can see that boundary scores from both matrices (“Boundary score 1” and “Boundary score 2”) frequently detected as significant boundaries in both matrices (boundary scores above the 1.5 threshold), but are non-differential (black dots). The differential boundaries correspond to the “Differential boundary score” track, where absolute boundary score differences above the 2.0 threshold are detected. The different types of differential boundaries are defined according to a schema described in the <code>TADCompare</code> manuscript. Note that coloring by types may be disabled by setting “show_types” to FALSE; only differential and non-differential labels will be shown.</p>
</div>
<div id="comparing-pre-defined-tad-boundaries" class="section level3">
<h3 class="hasAnchor">
<a href="#comparing-pre-defined-tad-boundaries" class="anchor"></a>Comparing pre-defined TAD boundaries</h3>
<p>We can also use pre-specified TADs by providing a list of TAD coordinates containing TAD boundaries for each contact matrix. The list should be of length 2. We show how to do this below, using SpectralTAD for pre-specification:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Call TADs using SpectralTAD</span>
<span class="va">bed_coords1</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpectralTAD/man/SpectralTAD.html">SpectralTAD</a></span><span class="op">(</span><span class="va">mtx1</span>, chr <span class="op">=</span> <span class="st">"chr2"</span>, levels <span class="op">=</span> <span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Estimating resolution</code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bed_coords2</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html">bind_rows</a></span><span class="op">(</span><span class="fu">SpectralTAD</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SpectralTAD/man/SpectralTAD.html">SpectralTAD</a></span><span class="op">(</span><span class="va">mtx2</span>, chr <span class="op">=</span> <span class="st">"chr2"</span>, levels <span class="op">=</span> <span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Estimating resolution</code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Placing the data in a list for the plotting procedure</span>
<span class="va">Combined_Bed</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">bed_coords1</span>, <span class="va">bed_coords2</span><span class="op">)</span>

<span class="co"># Running TADCompare with pre-specified TADs</span>
<span class="va">TD_Compare</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/TADCompare.html">TADCompare</a></span><span class="op">(</span><span class="va">mtx1</span>, <span class="va">mtx2</span>, resolution <span class="op">=</span> <span class="va">res</span>, pre_tads <span class="op">=</span> <span class="va">Combined_Bed</span><span class="op">)</span>
<span class="co"># Running the plotting algorithm with pre-specified TADs</span>
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DiffPlot.html">DiffPlot</a></span><span class="op">(</span>tad_diff    <span class="op">=</span> <span class="va">TD_Compare</span>, 
              cont_mat1   <span class="op">=</span> <span class="va">mtx1</span>,
              cont_mat2   <span class="op">=</span> <span class="va">mtx2</span>,
              resolution  <span class="op">=</span> <span class="va">res</span>,
              start_coord <span class="op">=</span> <span class="va">start_coord</span>,
              end_coord   <span class="op">=</span> <span class="va">end_coord</span>,
              pre_tad     <span class="op">=</span> <span class="va">Combined_Bed</span>,
              show_types  <span class="op">=</span> <span class="cn">FALSE</span>, 
              point_size  <span class="op">=</span> <span class="fl">5</span>,
              max_height  <span class="op">=</span> <span class="fl">10</span>,
              rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1.5</span>, <span class="fl">2</span><span class="op">)</span>,
              palette     <span class="op">=</span> <span class="st">"RdYlBu"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></code></pre></div>
<p><img src="TADCompare_files/figure-html/unnamed-chunk-11-1.png" width="960"></p>
<p>As we can see, the pre-specification of TADs allows us to outline TAD boundaries enhancing visualization. Additionally, the differential boundaries presented now correspond to those called by the TAD caller as opposed to those detected by <code>TADCompare</code> using boundary scores. Note that the pre-specified TAD boundaries do not necessarily correspond to the boundary scores; consequently, the classification schema is applied with respect to the pre-specified TAD boundaries. Howevery, using pre-specified TADs makes it is easier to visually interpret the differences of the differences. Therefore, we recommend setting “show_types” to FALSE. Additionally, for the pre-specified TADs, a new category is introduced, called “Non-Overlap.” Non-Overlap corresponds to boundaries that are determined to be non-differential by <code>TADCompare</code> but do not overlap when called by the TAD caller.</p>
</div>
</div>
</div>
<div id="timecompare" class="section level1">
<h1 class="hasAnchor">
<a href="#timecompare" class="anchor"></a>TimeCompare</h1>
<div id="introduction-1" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction-1" class="anchor"></a>Introduction</h2>
<p><code>TimeCompare</code> is a function for the time-course analysis of data. Briefly, a user inputs a list of contact matrices representing at least four time points. The <code>TimeCompare</code> function will run with fewer time points, but the classification of temporal changes may be incorrect. The output is a data frame containing all regions with TAD boundaries detected in at least one time point. These regions are further classified into six separate types of temporal changes (Highly common, dynamic, early/late appearing, and early/late disappearing) based on how TAD boundaries evolve over time. The function also returns a plot summarizing the occurrence of each TAD boundary and another data frame containing a summary of change at each region, regardless of whether a boundary was detected.</p>
</div>
<div id="running-timecompare" class="section level2">
<h2 class="hasAnchor">
<a href="#running-timecompare" class="anchor"></a>Running TimeCompare</h2>
<p><code>TimeCompare</code> function takes a list of matrices similar in format to <code>TADCompare</code> function. Like <code>TADCompare</code>, <code>TimeCompare</code> function will estimate resolution and convert matrices to the appropriate format. For this example, we use time-varying sparse 3-column contact matrices from <span class="citation">(Suhas S. P. Rao et al. 2017)</span>, representing the HCT-116 cell line. The matrices represent a single chromosome 22 cell sample treated with TAD boundary destroying auxin. The data is then sequenced at four time points (20, 40, 60, and 180 minutes) after the withdrawal of auxin. Once auxin is withdrawn, the TAD boundaries slowly return. Using <code>TimeCompare</code> function, we can track the return of TADs after withdrawal.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get the list of contact matrices</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"time_mats"</span><span class="op">)</span>
<span class="co"># Checking format</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">time_mats</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 3
##         X1       X2    X3
##      &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;
## 1 16050000 16050000    14
## 2 16100000 16100000     3
## 3 16200000 16200000     1
## 4 16250000 16250000     1
## 5 16300000 16300000    10
## 6 16350000 16350000     2</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># These are sparse 3-column matrices</span>
<span class="co"># Running MultiCompare</span>
<span class="va">time_var</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TimeCompare.html">TimeCompare</a></span><span class="op">(</span><span class="va">time_mats</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span></code></pre></div>
<pre><code>## Converting to n x n matrix</code></pre>
<pre><code>## Matrix dimensions: 704x704
## Matrix dimensions: 704x704
## Matrix dimensions: 704x704
## Matrix dimensions: 704x704</code></pre>
<p>The first item returned by <code>TimeCompare</code> function is <code>TAD_Bounds</code>, a data frame containing all regions that contain a TAD boundary detected in at least one time point:</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">time_var</span><span class="op">$</span><span class="va">TAD_Bounds</span><span class="op">)</span></code></pre></div>
<pre><code>##   Coordinate   Sample 1   Sample 2   Sample 3   Sample 4 Consensus_Score
## 1   16900000 -0.6733709 -0.7751516 -0.7653696 15.1272252     -0.71937026
## 2   17350000  3.6406563  2.3436229  3.0253018  0.7840556      2.68446232
## 3   18850000  0.6372268  6.3662244 -0.7876844  6.9255446      3.50172562
## 4   20700000  1.5667926  3.0968633  2.9130479  2.8300136      2.87153075
## 5   22000000 -1.0079676 -0.7982571  0.6007264  3.1909178     -0.09876534
## 6   22050000 -1.0405532 -0.9892242 -0.2675822  4.2737511     -0.62840320
##                 Category
## 1     Late Appearing TAD
## 2 Early Disappearing TAD
## 3    Early Appearing TAD
## 4            Dynamic TAD
## 5     Late Appearing TAD
## 6     Late Appearing TAD</code></pre>
<p>The first column corresponds to genomic coordinates. The columns beginning with the “Sample” prefix correspond to the boundary score at the given coordinate in each sample. The consensus score is simply the median score across all samples, and the category corresponds to the type of change.</p>
<p><code>All_Bounds</code> is the second list entry and is identical in structure to the <code>TAD_Bounds</code> data frame, but it includes every region of the genome regardless of whether it is a TAD or not.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">time_var</span><span class="op">$</span><span class="va">All_Bounds</span><span class="op">)</span></code></pre></div>
<pre><code>##   Coordinate   Sample 1   Sample 2   Sample 3   Sample 4 Consensus_Score
## 1   16900000 -0.6733709 -0.7751516 -0.7653696 15.1272252      -0.7193703
## 2   16950000 -0.6343671 -0.6624276 -0.4107173 -0.6656620      -0.6483973
## 3   17000000 -1.0409280 -0.7618272  0.7910678 -1.0008876      -0.8813574
## 4   17050000 -0.9199917 -0.7123180  0.6712399 -0.8254422      -0.7688801
## 5   17100000 -0.6097913 -0.4785485 -0.6232367 -0.2095729      -0.5441699
## 6   17150000 -0.7215028 -0.5669957  0.2413230 -0.6332710      -0.6001333
##             Category
## 1 Late Appearing TAD
## 2  Highly Common TAD
## 3  Highly Common TAD
## 4  Highly Common TAD
## 5  Highly Common TAD
## 6  Highly Common TAD</code></pre>
<p>We also include a stacked barplot that includes the number of times each type of temporal boundary occurs in the data. This plot is created in ggplot2 and fully customizable.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">time_var</span><span class="op">$</span><span class="va">Count_Plot</span></code></pre></div>
<p><img src="TADCompare_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
</div>
<div id="consensustads" class="section level1">
<h1 class="hasAnchor">
<a href="#consensustads" class="anchor"></a>ConsensusTAD</h1>
<div id="introduction-2" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction-2" class="anchor"></a>Introduction</h2>
<p><code>ConsensusTADs</code> function implements an approach for calling TADs across multiple datasets. It uses the median boundary score across <span class="math inline">\(n\)</span> replicates or cell lines to calculate a consensus of TAD boundaries, that is, consistently detected across replicates. This effectively filters out noisy TAD boundaries that may appear in only one or a few replicates or cell lines.</p>
</div>
<div id="running-consensustads" class="section level2">
<h2 class="hasAnchor">
<a href="#running-consensustads" class="anchor"></a>Running ConsensusTADs</h2>
<p><code>ConsensusTADs</code> function takes essentially the same input as the <code>TimeCompare</code> function (a list of contact matrices). It provides consensus TAD scores for each region, summarized across each contact matrix of the genome. It also provides a list of regions with significant TAD scores. These regions can be thought of as consensus TAD boundaries. Using these, we can get a single set of TADs summarized across a set of replicates, conditions, or time points. For this example, we use the two replicates from <span class="citation">(Suhas S. P. Rao et al. 2014)</span>.</p>
<p>We demonstrate how to run <code>ConsensusTADs</code> by calling consensus TADs on time-varying contact matrices created by treating a single sample with auxin, which destroys its TAD boundaries, and then tracking their return at four time points (20, 40, 60 and 180 minutes) <span class="citation">(Suhas S. P. Rao et al. 2017)</span>. The consensus boundary score is intended to provide a summary of TAD boundaries across all time points.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get the rao contact matrices built into the package</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"time_mats"</span><span class="op">)</span>
<span class="co"># Running MultiCompare</span>
<span class="va">con_tads</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ConsensusTADs.html">ConsensusTADs</a></span><span class="op">(</span><span class="va">time_mats</span>, resolution <span class="op">=</span> <span class="fl">50000</span><span class="op">)</span></code></pre></div>
<pre><code>## Converting to n x n matrix</code></pre>
<pre><code>## Matrix dimensions: 704x704
## Matrix dimensions: 704x704
## Matrix dimensions: 704x704
## Matrix dimensions: 704x704</code></pre>
<p><code>ConsensusTADs</code> returns two data frames. The first data frame, <code>Consensus</code>, contains all regions containing consensus TADs based on the consensus score.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">con_tads</span><span class="op">$</span><span class="va">Consensus</span><span class="op">)</span></code></pre></div>
<pre><code>##   Coordinate   Sample 1   Sample 2   Sample 3   Sample 4 Consensus_Score
## 1   16900000 -0.6733709 -0.7751516 -0.7653696 15.1272252     -0.71937026
## 2   17350000  3.6406563  2.3436229  3.0253018  0.7840556      2.68446232
## 3   18850000  0.6372268  6.3662244 -0.7876844  6.9255446      3.50172562
## 4   20700000  1.5667926  3.0968633  2.9130479  2.8300136      2.87153075
## 5   22000000 -1.0079676 -0.7982571  0.6007264  3.1909178     -0.09876534
## 6   22050000 -1.0405532 -0.9892242 -0.2675822  4.2737511     -0.62840320</code></pre>
<p>The columns correspond to the coordinate of the region with a significant boundary score, the individual boundary score for each region, and the consensus score.</p>
<p>The second data frame, <code>All_Regions</code>, is identical to the <code>Consensus</code> data frame, but it includes every region of the genome, which occurs in both matrices.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">con_tads</span><span class="op">$</span><span class="va">All_Regions</span><span class="op">)</span></code></pre></div>
<pre><code>##   Coordinate   Sample 1   Sample 2   Sample 3   Sample 4 Consensus_Score
## 1   16900000 -0.6733709 -0.7751516 -0.7653696 15.1272252      -0.7193703
## 2   16950000 -0.6343671 -0.6624276 -0.4107173 -0.6656620      -0.6483973
## 3   17000000 -1.0409280 -0.7618272  0.7910678 -1.0008876      -0.8813574
## 4   17050000 -0.9199917 -0.7123180  0.6712399 -0.8254422      -0.7688801
## 5   17100000 -0.6097913 -0.4785485 -0.6232367 -0.2095729      -0.5441699
## 6   17150000 -0.7215028 -0.5669957  0.2413230 -0.6332710      -0.6001333</code></pre>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS  10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] TADCompare_1.1.1  SpectralTAD_1.4.0 dplyr_1.0.2       BiocStyle_2.16.1 
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-0            ggsignif_0.6.0             
##   [3] ellipsis_0.3.1              rio_0.5.16                 
##   [5] CGHcall_2.50.0              rprojroot_2.0.2            
##   [7] DNAcopy_1.62.0              XVector_0.28.0             
##   [9] GenomicRanges_1.40.0        fs_1.5.0                   
##  [11] rstudioapi_0.13             farver_2.0.3               
##  [13] ggpubr_0.4.0                listenv_0.8.0              
##  [15] fansi_0.4.1                 HiCcompare_1.10.0          
##  [17] codetools_0.2-16            splines_4.0.2              
##  [19] R.methodsS3_1.8.1           impute_1.62.0              
##  [21] knitr_1.30                  Rsamtools_2.4.0            
##  [23] broom_0.7.2                 cluster_2.1.0              
##  [25] R.oo_1.24.0                 pheatmap_1.0.12            
##  [27] BiocManager_1.30.10         compiler_4.0.2             
##  [29] backports_1.2.0             assertthat_0.2.1           
##  [31] Matrix_1.2-18               cli_2.2.0                  
##  [33] limma_3.44.3                htmltools_0.5.0            
##  [35] tools_4.0.2                 gtable_0.3.0               
##  [37] glue_1.4.2                  GenomeInfoDbData_1.2.3     
##  [39] reshape2_1.4.4              Rcpp_1.0.5                 
##  [41] carData_3.0-4               Biobase_2.48.0             
##  [43] cellranger_1.1.0            pkgdown_1.6.1              
##  [45] vctrs_0.3.5                 Biostrings_2.56.0          
##  [47] nlme_3.1-150                QDNAseq_1.24.0             
##  [49] xfun_0.19                   stringr_1.4.0              
##  [51] globals_0.13.1              openxlsx_4.2.3             
##  [53] lifecycle_0.2.0             gtools_3.8.2               
##  [55] rstatix_0.6.0               InteractionSet_1.16.0      
##  [57] future_1.20.1               zlibbioc_1.34.0            
##  [59] scales_1.1.1                hms_0.5.3                  
##  [61] ragg_0.4.0                  parallel_4.0.2             
##  [63] SummarizedExperiment_1.18.2 rhdf5_2.32.4               
##  [65] RColorBrewer_1.1-2          yaml_2.2.1                 
##  [67] curl_4.3                    memoise_1.1.0              
##  [69] gridExtra_2.3               ggplot2_3.3.2              
##  [71] CGHbase_1.48.0              stringi_1.5.3              
##  [73] S4Vectors_0.26.1            desc_1.2.0                 
##  [75] BiocGenerics_0.34.0         zip_2.1.1                  
##  [77] BiocParallel_1.22.0         GenomeInfoDb_1.24.2        
##  [79] rlang_0.4.8                 pkgconfig_2.0.3            
##  [81] systemfonts_0.3.2           matrixStats_0.57.0         
##  [83] bitops_1.0-6                evaluate_0.14              
##  [85] lattice_0.20-41             purrr_0.3.4                
##  [87] Rhdf5lib_1.10.1             labeling_0.4.2             
##  [89] cowplot_1.1.0               tidyselect_1.1.0           
##  [91] parallelly_1.21.0           plyr_1.8.6                 
##  [93] magrittr_2.0.1              bookdown_0.21              
##  [95] R6_2.5.0                    IRanges_2.22.2             
##  [97] generics_0.1.0              DelayedArray_0.14.1        
##  [99] pillar_1.4.7                haven_2.3.1                
## [101] foreign_0.8-80              mgcv_1.8-33                
## [103] abind_1.4-5                 RCurl_1.98-1.2             
## [105] tibble_3.0.4                future.apply_1.6.0         
## [107] PRIMME_3.1-3                crayon_1.3.4               
## [109] car_3.0-10                  utf8_1.1.4                 
## [111] KernSmooth_2.23-18          rmarkdown_2.5              
## [113] readxl_1.3.1                grid_4.0.2                 
## [115] data.table_1.13.2           marray_1.66.0              
## [117] forcats_0.5.0               digest_0.6.27              
## [119] tidyr_1.1.2                 R.utils_2.10.1             
## [121] textshaping_0.2.1           stats4_4.0.2               
## [123] munsell_0.5.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Cresswell:2019aa" class="csl-entry">
Cresswell, Kellen G, John C Stansfield, and Mikhail G Dozmorov. 2019. <span>“SpectralTAD: An r Package for Defining a Hierarchy of Topologically Associated Domains Using Spectral Clustering.”</span> <em>bioRxiv</em>, January, 549170. <a href="https://doi.org/10.1101/549170">https://doi.org/10.1101/549170</a>.
</div>
<div id="ref-Rao:2017aa" class="csl-entry">
Rao, Suhas S. P., Su-Chen Huang, Brian Glenn St Hilaire, Jesse M. Engreitz, Elizabeth M. Perez, Kyong-Rim Kieffer-Kwon, Adrian L. Sanborn, et al. 2017. <span>“Cohesin Loss Eliminates All Loop Domains.”</span> <em>Cell</em> 171 (2): 305–320.e24. <a href="https://doi.org/10.1016/j.cell.2017.09.026">https://doi.org/10.1016/j.cell.2017.09.026</a>.
</div>
<div id="ref-Rao:2014aa" class="csl-entry">
Rao, Suhas S.P., Miriam H. Huntley, Neva C. Durand, Elena K. Stamenova, Ivan D. Bochkov, James T. Robinson, Adrian L. Sanborn, et al. 2014. <span>“A 3d Map of the Human Genome at Kilobase Resolution Reveals Principles of Chromatin Looping.”</span> <em>Cell</em> 159 (7): 1665–80. <a href="https://doi.org/10.1016/j.cell.2014.11.021">https://doi.org/10.1016/j.cell.2014.11.021</a>.
</div>
<div id="ref-Schmitt:2016aa" class="csl-entry">
Schmitt, Anthony D, Ming Hu, Inkyung Jung, Zheng Xu, Yunjiang Qiu, Catherine L Tan, Yun Li, et al. 2016. <span>“A Compendium of Chromatin Contact Maps Reveals Spatially Active Regions in the Human Genome.”</span> <em>Cell Rep</em> 17 (8): 2042–59. <a href="https://doi.org/10.1016/j.celrep.2016.10.061">https://doi.org/10.1016/j.celrep.2016.10.061</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kellen Cresswell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
